docs/hooks/lua.md

```diff
@@ -317,6 +317,63 @@ The `value` string should be in [ISO8601](https://en.wikipedia.org/wiki/ISO_8601
 
 Returns a new 128-bit [RFC 4122 UUID](https://www.rfc-editor.org/rfc/rfc4122){: target="_blank" } in string representation.
 
+### `net/http` (optional)
+
+Provides a `request` function that performs an HTTP request.
+For security reasons, this package is not available by default as it enables http requests to be sent out from the lakeFS instance network. The feature should be enabled under `actions.lua.net_http_enabled` [configuration](../reference/configuration.md).
+Request will time out after 30 seconds.
+
+```lua
+http.request(url [, body])
+http.request{
+  url = string,
+  [method = string,]
+  [headers = header-table,]
+  [body = string,]
+}
+```
+
+Returns a code (number), body (string), headers (table) and status (string).
+
+ - code - status code number
+ - body - string with the response body
+ - headers - table with the response request headers (key/value or table of values)
+ - status - status code text
```

Comment:
Modeled based on existing package, but with Go implementation - https://w3.impa.br/~diego/software/luasocket/http.html.
Usually the first 2 are enough to get by - the headers are next in case you need to process specific requests with authorizations.
The status text is usually a message that usually is used for extra information for messages in case of non success status codes.

Reply:
According to the documentation you sent:

> In case of failure, the function returns nil followed by an error message. If successful, the simple form returns the response body as a string, followed by the response status code, the response headers and the response status line. The generic function returns the same information, except the first return value is just the number 1 (the body goes to the sink).

Maybe we need to change to order after all?

Reply:
Regarding the return values - in case of an error we use exception. If the http call completes all the return values will be populated. So the only difference between us and the socket.http implementation will be that we return the code first.
I thought it will have more sense as it will be more relevant for cases where we do not use the body.